spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/LocalApiKeyManager.java

```diff
@@ -0,0 +1,122 @@
+package eu.coatrack.proxy.security;
+
+/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.api.ApiKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.net.ConnectException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Locally stores, periodically refreshes and provides API keys and their
+ * associated service APIs belonging to the gateway.
+ *
+ * @author Christoph Baier
+ */
+
+@EnableAsync
+@Service
+public class LocalApiKeyManager {
+
+    private static final Logger log = LoggerFactory.getLogger(LocalApiKeyManager.class);
+
+    private List<ApiKey> localApiKeyList = new ArrayList<>();
+    private LocalDateTime latestLocalApiKeyListUpdate = LocalDateTime.now();
+
+    private final ApiKeyFetcher apiKeyFetcher;
+    private final long numberOfMinutesTheGatewayShallWorkWithoutConnectionToAdmin;
+
+    public LocalApiKeyManager(
+            ApiKeyFetcher apiKeyFetcher,
+            @Value("${number-of-minutes-the-gateway-works-without-connection-to-admin}") long minutes) {
+        this.apiKeyFetcher = apiKeyFetcher;
+        this.numberOfMinutesTheGatewayShallWorkWithoutConnectionToAdmin = minutes;
+    }
+
+    public ApiKey findApiKeyFromLocalApiKeyList(String apiKeyValue) {
```

Comment:
A clearer name would be: 
```suggestion
    public ApiKey getApiKeyEntityByApiKeyValue(String apiKeyValue) {
```

Reply:
Done.